File: src/main/java/world/biome/BiomeManager.java
package world.biome;

import java.util.ArrayList;
import java.util.List;

public class BiomeManager {
    public static final BiomeManager INSTANCE = new BiomeManager();
    private static final List<BiomeBase> BIOMES = new ArrayList<>();

    private BiomeManager() {
        BIOMES.add(new BiomePlains());
    }

    public void registerBiome(BiomeBase base) {
        base.setID(BIOMES.size());
        BIOMES.add(base);
    }
}

File: src/main/kotlin/world/biome/BiomeManager.kt
package world.biome

object BiomeManager {
    val biomes: MutableList<BiomeBase> = mutableListOf()

    init {
        biomes.add(BiomePlains())
    }

    fun addBiome(biome: BiomeBase) {
        biome.ID = biomes.count()
        biomes.add(biome)
    }
}